src/VecSim/spaces/L2/L2_AVX512F_BW_VL_VNNI_UINT8.h

```diff
@@ -0,0 +1,78 @@
+/*
+ *Copyright Redis Ltd. 2021 - present
+ *Licensed under your choice of the Redis Source Available License 2.0 (RSALv2) or
+ *the Server Side Public License v1 (SSPLv1).
+ */
+
+#include "VecSim/spaces/space_includes.h"
+
+static inline void L2SqrStep(uint8_t *&pVect1, uint8_t *&pVect2, __m512i &sum) {
+    __m512i va = _mm512_loadu_epi8(pVect1); // AVX512BW
+    pVect1 += 64;
+
+    __m512i vb = _mm512_loadu_epi8(pVect2); // AVX512BW
+    pVect2 += 64;
+
+    __m512i va_lo = _mm512_unpacklo_epi8(va, _mm512_setzero_si512()); // AVX512BW
+    __m512i vb_lo = _mm512_unpacklo_epi8(vb, _mm512_setzero_si512());
+    __m512i diff_lo = _mm512_sub_epi16(va_lo, vb_lo);
+    sum = _mm512_dpwssd_epi32(sum, diff_lo, diff_lo);
+
+    __m512i va_hi = _mm512_unpackhi_epi8(va, _mm512_setzero_si512()); // AVX512BW
+    __m512i vb_hi = _mm512_unpackhi_epi8(vb, _mm512_setzero_si512());
+    __m512i diff_hi = _mm512_sub_epi16(va_hi, vb_hi);
+    sum = _mm512_dpwssd_epi32(sum, diff_hi, diff_hi);
+
+    // _mm512_dpwssd_epi32(src, a, b)
+    // Multiply groups of 2 adjacent pairs of signed 16-bit integers in `a` with corresponding
+    // 16-bit integers in `b`, producing 2 intermediate signed 32-bit results. Sum these 2 results
```

Comment:
Is there an unsigned equivalent for `_mm512_dpwssd_epi32`?

Reply:
No, but we don't need (or want) one.
We cast the `uint8_t` elements into **signed** `int16_t` (still with zero extension), and then we want to perform signed arithmetic operations.
for example, if we have elements 0 and 200, we want the result to be `(0-200)^2 = -200^2 = 40,000`, and not `(0-200)^2 = (uint16_t)(-200)^2 = 65,336^2 = 4,268,792,896`